src/Microsoft.DotNet.Cli.Utils/MarkOfTheWebDetector.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Win32.SafeHandles;
+using System;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.PlatformAbstractions;
+using RuntimeEnvironment = Microsoft.DotNet.PlatformAbstractions.RuntimeEnvironment;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    internal class MarkOfTheWebDetector : IMarkOfTheWebDetector
+    {
+        private const string ZoneIdentifierStreamName = "Zone.Identifier";
+        private const string ZoneIdIs3String = "ZoneId=3";
+
+        public bool HasMarkOfTheWeb(string filePath)
+        {
+            if (!File.Exists(filePath))
+            {
+                throw new FileNotFoundException($"{filePath} cannot be found");
+            }
```

Comment:
Generally, I would avoid an up-front check like this because it will be a race anyway. Is this to ensure that it throws on non-Windows instead of always returning false?

Reply:
Conclusion is to remove this check